solr/core/src/java/org/apache/solr/filestore/PackageStoreAPI.java

```diff
@@ -131,6 +131,53 @@ public void validateFiles(List<String> files, boolean validateSignatures, Consum
 
     static final String TMP_ZK_NODE = "/packageStoreWriteInProgress";
 
+    @EndPoint(
+            path = "/cluster/files/*",
+            method = SolrRequest.METHOD.DELETE,
+            permission = PermissionNameProvider.Name.FILESTORE_WRITE_PERM)
+    public void delete(SolrQueryRequest req, SolrQueryResponse rsp) {
+      if (!coreContainer.getPackageLoader().getPackageAPI().isEnabled()) {
+        throw new RuntimeException(PackageAPI.ERR_MSG);
+      }
+
+      try {
+        coreContainer.getZkController().getZkClient().create(TMP_ZK_NODE, "true".getBytes(UTF_8),
```

Comment:
do we use this for locking? no no no. we should be using zk versions for that.

I understand that the code may have been in place before, but we need to fix it then before we can add additional features on top of this. it will add so much trouble.

Reply:
I need to research how to do this better. I did not like this portion of the change but could not find a reliable way to check because of eventual consistency.

Reply:
@madrob This should be a separate ticket because this is used all over.

I've opened [SOLR-14698](https://issues.apache.org/jira/browse/SOLR-14698)

Reply:
thank you @noblepaul . I couldn't figure out a better way to do this other then simplifying the exception and conditional